### PR TITLE
Home feed

### PR DESCRIPTION
**Pull request by JakeSteinebronn**
# Changes
I won't put images here since it's just a feed of posts. On development, there's a button that enables the screen so we aren't slamming it every time we open the app, but it doesn't show up on a prod build. 

# Issue ticket number and link
#191

### CODE REVIEW

File: screens/home/HomeFeed.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      )}
+    </Center>
+  );
+
+  const renderSpinner = () => {
+    if (!exhausted) return <PostSkeleton />;
+    else return null;
+  };
+
+  const renderDivider = () => <Divider />;
```
Comment (tylerhm) [inline screens/home/HomeFeed.tsx:68]: This can be omitted, as the `Divider` is a function already.

File: screens/home/HomeFeed.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+  return (
+    <FlatList
+      ListHeaderComponent={header}
+      data={posts}
+      extraData={cursor}
+      onEndReached={getNextPosts}
+      ItemSeparatorComponent={renderDivider}
```
Comment (tylerhm) [inline screens/home/HomeFeed.tsx:76]:
```suggestion
      ItemSeparatorComponent={Divider}
```

**Review by tylerhm — COMMENTED** [review]
Looks like the `xplat` version isn't up to date. It asks for some functions that don't exist.
```sh
iOS Bundling failed 2164ms
Unable to resolve "../../xplat/queries/feed" from "screens/home/HomeFeed.tsx"
```

File: screens/home/HomeFeed.tsx (hunk 1 above)
Comment (JakeSteinebronn) [reply screens/home/HomeFeed.tsx:68]: oh yeah, I was gonna add spacing on the divider itself and forgot

File: screens/home/HomeFeed.tsx [hunk 3]
```diff
@@ -18,44 +20,63 @@ const HomeFeed = () => {
...
-    </>
+    <FlatList
+      ListHeaderComponent={header}
+      data={posts}
+      extraData={cursor}
+      onEndReached={getNextPosts}
+      ItemSeparatorComponent={renderDivider}
+      ListFooterComponent={renderSpinner}
+      renderItem={({ item }) => <Post post={item} />}
+    />
```
Comment (abrugal) [inline screens/home/HomeFeed.tsx:79]: Should add the key extractor

File: screens/home/HomeFeed.tsx [hunk 4]
```diff
@@ -18,44 +20,63 @@ const HomeFeed = () => {
     process.env.NODE_ENV !== 'development'
   );
   const [cursor, setCursor] = useState<PostCursorMerger>();
+  const [exhausted, setExhausted] = useState<boolean>(false);
   const [posts, setPosts] = useState<PostObj[]>([]);
 
   useEffect(() => {
     if (enabled && signedInUser !== undefined && userQuery.data !== undefined) {
-      console.log("Ok let's get it goin!");
       const newCursor = getQueryCursor(userQuery.data.followingList);
```
Comment (abrugal) [inline screens/home/HomeFeed.tsx:28]: Will this only show Posts from people the user is following? There should also be an option to see posts from everyone, even if you don't follow them. Or will that be implemented in a different PR?

Comment (JakeSteinebronn) [reply screens/home/HomeFeed.tsx:28]: This is currently only posts for whom  you're following. I didn't know we were doing posts from everyone; that can be a different PR

File: screens/home/HomeFeed.tsx [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+      ListHeaderComponent={header}
+      data={posts}
+      extraData={cursor}
+      onEndReached={getNextPosts}
+      ItemSeparatorComponent={Divider}
+      ListFooterComponent={renderSpinner}
+      renderItem={({ item }) => (
+        <Center>
+          <Post post={item} />
+        </Center>
```
Comment (tylerhm) [inline screens/home/HomeFeed.tsx:96]:
```suggestion
        <Box my={2}>
          <Post post={item} />
        </Box>
```
Lets switch to `Box` add some padding here so that we don't have super squished content

image

File: screens/home/HomeFeed.tsx [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+  useEffect(() => {
+    if (cursor) getNextPosts();
+  }, [cursor, getNextPosts]);
+
+  const header = () => (
+    <Center>
+      {enabled ? null : (
+        <Button onPress={() => setEnabled(true)}>Enable</Button>
+      )}
+    </Center>
```
Comment (tylerhm) [inline screens/home/HomeFeed.tsx:69]:
```suggestion
    <Box>
      {enabled ? null : (
        <Button onPress={() => setEnabled(true)}>Enable</Button>
      )}
    </Box>
```

File: screens/home/HomeFeed.tsx [hunk 7]
```diff
@@ -0,0 +1,103 @@
+import {
+  Button,
+  Center,
```
Comment (tylerhm) [inline screens/home/HomeFeed.tsx:3]:
```suggestion
  Box,
```
Native components automatically take up the full width that they can in a flexbox, and this is the behavior that we want for Posts. Although it works, `Center`ing them can lead to unexpected behavior and shrinkage.

**Review by tylerhm — APPROVED** [review]

**Comment by tylerhm** [comment]
Awesome, some small changes before merge. So excited to see it coming together :D
